src/Settings/Default.hs

```diff
@@ -147,27 +148,29 @@ testsuitePackages = do
 -- * We build 'dynamic' way when stage > Stage0 and the platform supports it.
 defaultLibraryWays :: Ways
 defaultLibraryWays = mconcat
-    [ pure [vanilla]
-    , notStage0 ? pure [profiling]
-    -- , notStage0 ? platformSupportsSharedLibs ? pure [dynamic]
+    [ vanillaAlways
+    , notStage0 ? profilingAlways
+    , notStage0 ? dynamicWhenPossible
     ]
 
 -- | Default build ways for the RTS.
 defaultRtsWays :: Ways
-defaultRtsWays = do
-    ways <- getLibraryWays
-    mconcat
-        [ pure [ logging, debug, threaded, threadedDebug, threadedLogging ]
-        , (profiling `elem` ways) ?
-          pure [ profiling, threadedProfiling, debugProfiling
-               , threadedDebugProfiling ]
-          -- we don't add the 'logging' variants of those for now, but we might
-          -- in the future?
+defaultRtsWays = mconcat
+  [ quickRtsWays
```

Comment:
Perhaps, we could reverse this and make `quickRtsWays` depend on `defaultRtsWays`, which seems more common.

Reply:
I mean, we usually want the default ways to be listed explicitly, and all derived ways to be expressed as deltas over the default one.